content/blog/2022-07-05-serving-models-with-mlem.md

```diff
@@ -0,0 +1,283 @@
+  model to an API.
+descriptionLong: >
+  Getting a machine learning model out to our end users can be an involved task.
+  You have to make sure it's in the correct format for the environment and for
+  the way it will be consumed. We're going to walk through how you can use MLEM
+  to save your model and serve it with a web API.
+picture: 2022-07-05/serve_mlem_models.png
+pictureComment: Serving Machine Models with MLEM
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/syncing-data-to-aws-s3/1192
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - MLEM
+  - Collaboration
+---
+
+Training a machine learning model is only one step in the process of getting
+something useful out to end-users. There are many goals for machine learning
+projects and we need accurate models that provide us with information based on
+the data received from a production environment. That could mean data coming
+from an API or batch processing, for example.
+
+Either way, you'll need to save your trained and validated model in a format
+that's consumable by other systems. That's why we'll be covering how to serve
+models through a [REST](https://restfulapi.net/) endpoint or a Python package
+with [MLEM](https://mlem.ai/).
+
+> You can get the repo we're working with
+> [here](https://github.com/iterative/stale-model-example/tree/mlem-serve).
+
+## Take a candidate model
+
+There are instructions in the project
+[README](https://github.com/iterative/stale-model-example/tree/mlem-serve#readme)
+on how to get everything you need installed and running. This is a simple ML
+project that uses DVC for data versioning and experiment tracking.
+
+After you have the repo set up, you'll already have the `MLEM` package
+installed. This project already has a model that's been trained and validated so
+we can move on to saving this model.
+
+## Save the model
+
+Inside the `train.py` script, we need to add the `MLEM` import to save the
+models as we experiment. We don't have to worry about running the training
+script for this project since we have the model, but it's good to know what's
+happening under the hood.
+
+```python
+# train.py
+import os
+import pickle5 as pickle
+import sys
+import yaml
+
+from mlem.api import save
+import numpy as np
+from sklearn.ensemble import RandomForestClassifier
+...
+```
+
+Then you can add the `save` function to the end of the training script.
+
+```python
+# train.py
+...
+clf = RandomForestClassifier(
+    n_estimators=n_est, min_samples_split=min_split, n_jobs=2, random_state=seed
+)
+
+clf.fit(x, labels)
+
+save(
+    clf,
+    "clf",
+    sample_data=x,
+    description="Random Forest Classifier",
+)
+```
+
+<admon type="tip">
+
+You don't have to do these steps as we already have a model available, but if
+you want to see the training and evaluation steps in action, you can run a few
+experiments with:
+
+```dvc
+$ dvc exp run
+```
+
+This will execute the DVC pipeline and you can check out what is happening in
```

Comment:
Other than the mention above (see my previous comment), I'm not sure we should mix in the concept of DVC experiments to this post, because more explaining would be required, which seems out of topic: how to update params, compare metrics/plots, etc.

````suggestion
you want to see the training and evaluation steps in action, you reproduce
the DVC pipeline with:

```dvc
$ dvc repro
```

You can check out what is happening in
````

There's a couple other mentions of DVC exps down in lines 139 and 143.

That said, if you do keep this maybe link `[experiments]` to https://dvc.org/doc/start/experiments (again) for context.